## Tak kenal Maka Tak Sayang

- 1973005
- Ferly Jeremi Purnawan Apiang

# Load Breast Cancer Dataset

In [5]:
from sklearn.datasets import load_breast_cancer

Breast_cancer = load_breast_cancer()

## Dataset Dimension

In [6]:
Breast_cancer['data'].shape

(569, 30)

Breast Cancer dataset memiliki 569 baris dan 30 kolom.

## Training Set dan Testing Set

Kita akan membagi dataset kedalam Training Set dan Testing Set nya.

In [7]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(
    cancer['data'], cancer['target'], random_state = 0)

print("X_train shape", format(X_train.shape))
print("X_test shape", format(X_test.shape))

print("Y_train shape", format(y_train.shape))
print("Y_test shape", format(y_test.shape))

X_train shape (426, 30)
X_test shape (143, 30)
Y_train shape (426,)
Y_test shape (143,)


- Pertama-tama kita import terlebih dahulu modulnya dengan memanggil *from sklearn.model_selection import train_test_split*.
- Lalu kita panggil *train_test_split* dengan menyertakan parameter *(cancer['data'], cancer['target'], random_state = 0)*.
- Lalu kita tampikan *X_train* dan *X_test* nya untuk melihat dimensi dari training set dan testing set nya.

Berdasarkan hasil output kita bisa lihat bahwa X_train shape (426, 30) dan X_test shape (143, 30).
    


## KNN Traning Classification

Model machine learning yang akan kita gunakan adalah **KNN**.

In [8]:
from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier(n_neighbors = 1)

knn.fit(X_train, y_train)

KNeighborsClassifier(n_neighbors=1)

- Import 'from sklearn.neighbors import KNeighborsClassifier'.
- Lalu kita panggil *KNeighborsClassifier* dengan menyertakan parameter *(n_neighbors = 1)* artinya kita menentukan jumlah tetangga pada kasus ini ialah 1 tetangga terdekat yang kita tampung ke dalam variabel *knn*.
- *X_train* dan *y_train* dengan memanggil *fit(X_train, y_train)* yang kita kenakan pada variabel *knn*. 

## Prediction Breast Cancer Dataset by KNN

Melakukan prediksi terhadap X_test yang dimiliki.

In [10]:
y_pred = knn.predict(X_test)

# Classification Report

In [11]:
from sklearn.metrics import classification_report

cls_report = classification_report(y_test, y_pred)

print(f'Classification Report :\n{cls_report}')

Classification Report :
              precision    recall  f1-score   support

           0       0.90      0.87      0.88        53
           1       0.92      0.94      0.93        90

    accuracy                           0.92       143
   macro avg       0.91      0.91      0.91       143
weighted avg       0.92      0.92      0.92       143



- Lakukan Import *'from sklearn.metrics import classification_report'*.
- Untuk selanjutnya kita akan memanggil *'classification_report'* dengan menyertakan parameter *'(y_test, y_pred)'* yang kita tampung kedalam variabel *cls_report*.

Merupakan *classification report* sebagai bentuk evaluasi terhadap model **KNN** pada **Breast Cancer** dataset yang baru saja kita training sebelumnya, ditampilkan nilai *precision*, *recall*, serta *f1-score* nya.